packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -274,28 +291,33 @@ class _DraggableSheetExtent {
     required this.snap,
     required this.snapSizes,
     required this.initialExtent,
-    required VoidCallback listener,
-  }) : assert(minExtent != null),
-       assert(maxExtent != null),
-       assert(initialExtent != null),
-       assert(minExtent >= 0),
-       assert(maxExtent <= 1),
-       assert(minExtent <= initialExtent),
-       assert(initialExtent <= maxExtent),
-       _currentExtent = ValueNotifier<double>(initialExtent)..addListener(listener),
-       availablePixels = double.infinity;
+    required this.onExtentChanged,
+    ValueNotifier<double>? currentExtent,
+    bool? hasChanged,
```

Comment:
Maybe it's early, but why can't this be `bool hasChanged = false`?

Reply:
I thought it was better to treat the widget has having changed even if there was a call to build with new parameters since the last drag.

If we use `hasChanged = false` here, we can't trigger a snap to pick up any changed snap configuration. TBF, it was just a value judgement on my part that changing snap configuration should trigger a snap-open to input on this decision. I thought it was desirable as, if the a developer is modifying the snap configuration, they probably want to see the modification take effect.

Additionally, `hasChanged = false` suggests that the sheet should be rebuilt at `initialExtent`. I've roughly modeled behavior here after `PageView` and `PageController.initialPage` where `initialPage` is only referenced on the first build even if the widget is rebuilt with a new initial page.

Reply:
SGTM